### PR TITLE
aeskeyfind: init at 1.0

### PR DESCRIPTION
**Pull request by FedX-sudo**
###### Description of changes
Init pentesting tool aeskeyfind. 

###### Things done
Add aeskeyfind another security tool and recommended to be used with my last PR aeskfix. 

Relevant issue: https://github.com/NixOS/nixpkgs/issues/81418

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "aeskeyfind";
+  version = "1.0";
+
+  src = fetchurl {
+    url = "https://citpsite.s3.amazonaws.com/memory-content/src/aeskeyfind-${version}.tar.gz";
+    sha256 = "sha256-FBflwbYehruVJ9sfW+4ZlaDuqCR12zy8iA4Ev3Bgg+Q=";
+  };
+  installPhase = ''
```
Comment (fabaff) [inline pkgs/tools/security/aeskeyfind/default.nix:14]:
```suggestion
  };

  installPhase = ''
```

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    runHook preInstall
+    mkdir -p $out/bin
+    cp aeskeyfind $out/bin
+    runHook postInstall
+  '';
+  meta = with lib; {
```
Comment (fabaff) [inline pkgs/tools/security/aeskeyfind/default.nix:20]:
```suggestion
  '';

  meta = with lib; {
```

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    description = "Locates 128-bit and 256-bit AES keys in a captured memory image.";
+    homepage = "https://citp.princeton.edu/our-work/memory/";
+    maintainers = with maintainers; [ fedx-sudo ];
+  };
+
```
Comment (fabaff) [inline pkgs/tools/security/aeskeyfind/default.nix:25]:
```suggestion
  };
```

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "sha256-FBflwbYehruVJ9sfW+4ZlaDuqCR12zy8iA4Ev3Bgg+Q=";
+  };
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp aeskeyfind $out/bin
+    runHook postInstall
+  '';
+  meta = with lib; {
+    description = "Locates 128-bit and 256-bit AES keys in a captured memory image.";
```
Comment (fabaff) [inline pkgs/tools/security/aeskeyfind/default.nix:21]:
```suggestion
    description = "Locates 128-bit and 256-bit AES keys in a captured memory image";
```

see https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    url = "https://citpsite.s3.amazonaws.com/memory-content/src/aeskeyfind-${version}.tar.gz";
+    sha256 = "sha256-FBflwbYehruVJ9sfW+4ZlaDuqCR12zy8iA4Ev3Bgg+Q=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    cp aeskeyfind $out/bin
+    runHook postInstall
+  '';
```
Comment (papojari) [inline pkgs/tools/security/aeskeyfind/default.nix:20]: Why is the program not being compiled but instead an already compiled binary is used?

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    description = "Locates 128-bit and 256-bit AES keys in a captured memory image";
+    homepage = "https://citp.princeton.edu/our-work/memory/";
+    maintainers = with maintainers; [ fedx-sudo ];
+  };
```
Comment (papojari) [inline pkgs/tools/security/aeskeyfind/default.nix:26]: What about `meta.license`?

**Review by auroraanna — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 6 above)
Comment (FedX-sudo) [reply pkgs/tools/security/aeskeyfind/default.nix:26]: There is no license that I can find.

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 5 above)
Comment (FedX-sudo) [reply pkgs/tools/security/aeskeyfind/default.nix:20]: I believe it is. I will look into this more latter today.

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 6 above)
Comment (papojari) [reply pkgs/tools/security/aeskeyfind/default.nix:26]: That's a problem.

Comment (papojari) [reply pkgs/tools/security/aeskeyfind/default.nix:26]:
```suggestion
  meta = with lib; {
    description = "Locates 128-bit and 256-bit AES keys in a captured memory image";
    homepage = "https://citp.princeton.edu/our-work/memory/";
    license = unfree;
    maintainers = with maintainers; [ fedx-sudo ];
  };
```

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 5 above)
Comment (papojari) [reply pkgs/tools/security/aeskeyfind/default.nix:20]: You're right, the binary is not in the source. So somehow it is being compiled. I guess that using to the Makefile is the default of the buildPhase. In conclusion this is solved.

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 6 above)
Comment (fabaff) [reply pkgs/tools/security/aeskeyfind/default.nix:26]: The source tarball contains a `LICENSE` file.

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 5 above)
Comment (fabaff) [reply pkgs/tools/security/aeskeyfind/default.nix:20]: `stdenv` provides `gcc` and `make`.

File: pkgs/tools/security/aeskeyfind/default.nix (hunk 6 above)
Comment (papojari) [reply pkgs/tools/security/aeskeyfind/default.nix:26]: Huh, I literally opened the tarball but didn't see the license file. Thanks.

Comment (papojari) [reply pkgs/tools/security/aeskeyfind/default.nix:26]:
```suggestion
  meta = with lib; {
    description = "Locates 128-bit and 256-bit AES keys in a captured memory image";
    homepage = "https://citp.princeton.edu/our-work/memory/";
    license = bsd3;
    maintainers = with maintainers; [ fedx-sudo ];
  };
```

**Review by auroraanna — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/aeskeyfind/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    runHook preInstall
+    mkdir -p $out/bin
+    cp aeskeyfind $out/bin
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Locates 128-bit and 256-bit AES keys in a captured memory image";
+    homepage = "https://citp.princeton.edu/our-work/memory/";
+    license = bsd3;
```
[the remark this answers is not in the document]
Comment (FedX-sudo) [reply pkgs/tools/security/aeskeyfind/default.nix]: Thanks!

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 166420` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>aeskeyfind</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/166420#issuecomment-1088362854)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...$ ./results/aeskeyfind/bin/aeskeyfind 
Usage: aeskeyfind [OPTION]... MEMORY-IMAGE
Locates scheduled 128-bit and 256-bit AES keys in MEMORY-IMAGE.

	-v		verbose output -- prints the extended keys and 
			the constraints on the rows of the key schedule
[...]
```

Looks good to me.

**Pull request merged**